sdks/java/core/src/test/java/org/apache/beam/sdk/util/FileIOChannelFactoryTest.java

```diff
@@ -194,19 +207,23 @@ public void testMatchWithDirectoryFiltersOutDirectory() throws Exception {
 
   @Test
   public void testResolve() throws Exception {
-    String expected = temporaryFolder.getRoot().toPath().resolve("aa").toString();
-    assertEquals(expected, factory.resolve(temporaryFolder.getRoot().toString(), "aa"));
+    Path rootPath = temporaryFolder.getRoot().toPath();
+    String rootString = rootPath.toString();
+
+    String expected = rootPath.resolve("aa").toString();
+    assertEquals(expected, factory.resolve(rootString, "aa"));
+    assertEquals(expected, factory.resolve("file://" + rootString, "aa"));
```

Comment:
Also add `factory.resolve("file:" + rootString, "aa")`

I'm not sure what URIs with `file:tmp/foo/bar` and `file://tmp/foo/bar` will resolve to. I expect that they'd be relative paths to the CWD of the JVM (which is something that should be extremely discouraged), but if so I don't know if it's worth the effort to check that the path is absolute rather than relative.